web/src/components/network/NetworkPage.jsx

```diff
@@ -58,135 +43,117 @@ const NoWiredConnections = () => {
 
-  useEffect(() => {
-    return client.onNetworkChange(({ type, payload }) => {
-      switch (type) {
-        case NetworkEventTypes.DEVICE_ADDED: {
-          setDevices((devs) => {
-            const newDevices = devs.filter((d) => d.name !== payload.name);
-            return [...newDevices, client.fromApiDevice(payload)];
-          });
-          break;
-        }
+  const fetchState = useCallback(async () => {
+    const devices = await client.devices();
+    const connections = await client.connections();
+    setDevices(devices);
+    setConnections(connections);
+  }, [client]);
 
-        case NetworkEventTypes.DEVICE_UPDATED: {
-          const [name, data] = payload;
-          setDevices(devs => {
-            const newDevices = devs.filter((d) => d.name !== name);
-            return [...newDevices, client.fromApiDevice(data)];
-          });
-          break;
-        }
+  useEffect(() => {
+    fetchState();
+    setUpdateState(false);
+  }, [fetchState, updateState]);
 
-        case NetworkEventTypes.DEVICE_REMOVED: {
-          setDevices(devs => devs.filter((d) => d.name !== payload));
-          break;
-        }
+  useEffect(() => {
+    return client.onNetworkChange(({ type }) => {
+      if ([NetworkEventTypes.DEVICE_ADDED, NetworkEventTypes.DEVICE_UPDATED, NetworkEventTypes.DEVICE_REMOVED].includes(type)) {
+        setUpdateState(true);
       }
-      client.connections().then(setConnections);
     });
-  }, [client, devices]);
-
-  useEffect(() => {
-    if (devices !== undefined) return;
+  });
 
-    client.devices().then(setDevices);
-  }, [client, devices]);
-
-  const selectConnection = ({ id }) => {
-    client.getConnection(id).then(setSelectedConnection);
-  };
+  const connectionDevice = ({ id }) => devices?.find(({ connection }) => id === connection);
+  const connectionAddresses = (connection) => {
+    const device = connectionDevice(connection);
+    const addresses = device ? device.addresses : connection.addresses;
 
-  const forgetConnection = async ({ id }) => {
-    await client.deleteConnection(id);
-    setConnections(undefined);
-  };
-
-  const updateConnections = async () => {
-    setConnections(undefined);
-    setDevices(undefined);
+    return addresses?.map(formatIp).join(", ");
   };
 
   const ready = (connections !== undefined) && (devices !== undefined);
 
   const WifiConnections = () => {
-    const wifiConnections = connections.filter(c => c.wireless);
+    const { wireless_enabled: wifiAvailable } = settings;
 
-    if (wifiConnections.length === 0) {
+    if (!wifiAvailable) {
       return (
-        <NoWifiConnections wifiScanSupported={settings.wireless_enabled} />
+        <CardField>
+          <CardField.Content>
+            <EmptyState title={_("No WiFi support")} icon="wifi_off" color="warning-color-200">
+              {_("The system does not support WiFi connections, probably because of missing or disabled hardware.")}
```

Comment:
Put in the shoes of an inexperienced user... I do not know if "missing hardware" is good enough.